aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -178,6 +179,19 @@ impl<T: Transaction, O: TransactionOutput<Txn = T>, E: Debug + Send + Clone>
         }
     }
 
+    pub(crate) fn output_approx_size(&self, txn_idx: TxnIndex) -> Option<u64> {
+        match &self.outputs[txn_idx as usize]
+            .load_full()
+            .expect("[BlockSTM]: Execution output must be recorded after execution")
+            .output_status
+        {
+            ExecutionStatus::Success(output) | ExecutionStatus::SkipRest(output) => {
+                Some(output.output_approx_size())
+            },
+            _ => None,
```

Comment:
It may be a refactor worthy to do separately, but I feel like we may have a pattern of forwarding on Success and SkipRest and some sort of default on the remaining ones. We could make this a macro / set of macros? This way, we also don't have to match _, and might simplify the code. Primarily asking if it makes sense to others - @igor-aptos - if yes, I suppose we don't do here - I can send a quick PR? let me know